Outreachy startup task for defaults.csv

Soniyanayak51 opened the pull request:
PR for issue #2 
1. Added module with all functions to import from jupyter notebook.
2. Added Jupyter Notebook with outputs.

I have used the following 2 models on the defaults.csv dataset:
1. Logistic Regression
2. SVM
Both using scikit-learn.

mlopatka commented on dev/Soniyanayak51/classificationModelFunctions.py:
```diff
@@ -0,0 +1,175 @@
+    '''
+    Loads data using pandas read_csv.
+    '''
+    df = pd.read_csv("/home/soniya51/Downloads/default of credit card clients.csv")
```
please split out the local filepath as a variable to make local review easier.

Soniyanayak51 replied:
@mlopatka Thanks for the review! 
Sure, I'll incorporate the suggested changes.

dzeber replied:
Also, please make sure to use the CSV file included in the repo. That way anyone who runs the code should get consistent results. This will require you change other things in the notebook that depend on the CSV structure, eg column headings.

Soniyanayak51 commented:
> This is a good start on the default dataset. A bit more interpretation of the results would be helpful.
> 
> * confusion_matrix` results are output, but not discussed.
> * data is cleaned to consolidate categories, have alternate strategies been considered?
> * what is the motivation for the two classifiers chosen `svm` and `logistic regression`?
>   Please extend the analysis to discuss the problem and interpret the results achieved.

Soniyanayak51 commented:
@mlopatka Thanks for the review!
Sure, I'll incorporate the suggested changes.

dzeber commented on dev/Soniyanayak51/classificationModelFunctions.py:
```diff
@@ -0,0 +1,175 @@
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+import tensorflow as tf
```
Please drop this import

dzeber commented on dev/Soniyanayak51/OutreachyStartupTaskSoniyanayak51.ipynb:
```diff
@@ -0,0 +1,983 @@
+    "import pandas as pd\n",
+    "import numpy as np\n",
+    "import matplotlib.pyplot as plt\n",
+    "import tensorflow as tf\n",
```
Please drop this import

dzeber commented on dev/Soniyanayak51/OutreachyStartupTaskSoniyanayak51.ipynb:
```diff
@@ -0,0 +1,983 @@
+    }
+   ],
+   "source": [
+    "tf.config.experimental.list_physical_devices('GPU') # check for GPU"
```
Please drop this cell

dzeber commented on dev/Soniyanayak51/classificationModelFunctions.py:
```diff
@@ -0,0 +1,175 @@
+    for column in x.columns:
+        mean = np.mean(x_train[column])
+        std = np.std(x_train[column])
+        x[column] = (x[column] - mean) / std
```
I believe this standardization can be done more easily using `StandardScaler`, which I see you imported above. I'd recommend using that unless there is a reason why it didn't work for you.

dzeber commented on dev/Soniyanayak51/OutreachyStartupTaskSoniyanayak51.ipynb:
```diff
@@ -0,0 +1,983 @@
+    "from pandas import set_option\n",
+    "from sklearn.svm import SVC\n",
+    "from sklearn.linear_model import LogisticRegression\n",
+    "from sklearn.datasets import load_iris"
```
Please drop any imports that are not explicitly used in the notebook.

Soniyanayak51 commented:
@dzeber thanks for the review! I'll update the PR with the suggested changes.

mlopatka commented:
@Soniyanayak51 Please let us know if you intend to continue with this PR. I see that some of the outstanding review comments have not yet been addressed. Please lag me with an `@` mention when you are ready for another pass of this PR.

Soniyanayak51 commented:
@mlopatka I was working on some other PR's but I have made the requested changes regarding standardization, imports and discussion. I have used the UCI default of credit card clients Data Set but have changed the xls file to csv. Kindly review and let me know if I need to elaborate more on this notebook. I'll be happy to make the changes!

Soniyanayak51 commented:
@mlopatka @dzeber please review.

dzeber commented:
> Also, please make sure to use the CSV file included in the repo. That way anyone who runs the code should get consistent results. This will require you change other things in the notebook that depend on the CSV structure, eg column headings.

You need to use the [CSV file](https://github.com/mozilla/PRESC/blob/master/datasets/defaults.csv) that is actually included in the repo. It has undergone minor reformatting and is slightly different from the raw UCI dataset, so it will require you to adjust some of your data processing code. Sorry if that wasn't clear!

Soniyanayak51 commented:
@mlopatka @dzeber I have added the dataset included in the repo and python black formatting. Also added axes lables and titles to graphs. Please review.

mlopatka commented:
`"../../.././datasets/defaults.csv"`
evaluates to the same as
"../../../datasets/defaults.csv"
Please take the time to understand the changes and test locally before pushing new commits.
